pkg/cloud/gcp/actuators/util/gcp_tags_labels.go

```diff
@@ -119,3 +175,135 @@ func GetLabelsList(userLabelsAllowed bool, client controllerclient.Client, machi
 
 	return labels, nil
 }
+
+// getTagClientOptions returns the tag client options adding the credentials and
+// the endpoint which will be used by the client.
+func getTagClientOptions(serviceAccountJSON string) []option.ClientOption {
+	return []option.ClientOption{
+		option.WithCredentialsJSON([]byte(serviceAccountJSON)),
+		option.WithEndpoint(fmt.Sprintf("https://%s", resourceManagerHostSubPath)),
+	}
+}
+
+// NewTagValuesClient returns the client to be used for fetching tag values.
+func NewTagValuesClient(ctx context.Context, serviceAccountJSON string) (TagValuesClient, error) {
+	client, err := rscmgr.NewTagValuesRESTClient(ctx, getTagClientOptions(serviceAccountJSON)...)
+	return &tagValuesClient{client}, err
+}
+
+// getInfraResourceTagsList returns the user-defined tags present in the
+// status sub-resource of Infrastructure.
+func getInfraResourceTagsList(platformStatus *configv1.PlatformStatus) []machinev1.ResourceManagerTag {
+	if platformStatus != nil && platformStatus.GCP != nil && platformStatus.GCP.ResourceTags != nil {
+		tags := make([]machinev1.ResourceManagerTag, len(platformStatus.GCP.ResourceTags))
+		for i, tag := range platformStatus.GCP.ResourceTags {
+			tags[i] = machinev1.ResourceManagerTag{
+				ParentID: tag.ParentID,
+				Key:      tag.Key,
+				Value:    tag.Value,
+			}
+		}
+		return tags
+	}
+	return nil
+}
+
+// GetNamespacedTagValue is a wrapper for rscmgr.GetNamespacedTagValue() method.
+func (c *tagValuesClient) GetNamespacedTagValue(ctx context.Context, req *rscmgrpb.GetNamespacedTagValueRequest) (*rscmgrpb.TagValue, error) {
+	return c.TagValuesClient.GetNamespacedTagValue(ctx, req)
```

Comment:
This is the interaction with GCP right? Is tag management part of their compute service or a separate service? We currently have all GCP API calls for compute service in https://github.com/openshift/machine-api-provider-gcp/blob/main/pkg/cloud/gcp/actuators/services/compute/computeservice.go, can we make an equivalent/follow the same pattern for the parts of this code that interact with google APIs, so that we have a clean separation of concern for what is logic to talk to google, vs our internal tag management code

Reply:
suggestion incorporated.
`google.golang.org/api/cloudresourcemanager/v1` didn't have support for tag related APIs earlier, which has been updated in v3. Code updated to keep it inline with other implementation.